src/main/java/net/minecraftforge/fluids/FluidStack.java

```diff
@@ -84,17 +86,46 @@ public static FluidStack loadFluidStackFromNBT(NBTTagCompound nbt)
         {
             return null;
         }
-        if (!nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+
+        FluidStack stack;
+        if (nbt.hasKey("Fluid", Constants.NBT.TAG_STRING))
         {
-            return null;
+            ResourceLocation fluid = new ResourceLocation(nbt.getString("Fluid"));
+            if (!ForgeRegistries.FLUIDS.containsKey(fluid))
+            {
+                return null;
+            }
+            else
+            {
+                stack = new FluidStack(ForgeRegistries.FLUIDS.getValue(fluid), nbt.getInteger("Amount"));
+            }
         }
-
-        String fluidName = nbt.getString("FluidName");
-        if (FluidRegistry.getFluid(fluidName) == null)
+        //TODO Drop support for this in 1.14 or 1.15
```

Comment:
Explain what this is for in the comment, so that future-whoever can decide if it's a good time to drop it.